Configuration/TCA/Overrides/static_countries.php

```diff
@@ -2,8 +2,6 @@
 
 declare(strict_types=1);
 
-defined('TYPO3_MODE') || die();
```

Comment:
As far as I can see, the TCA overrides should be protected with `defined('TYPO3') or die();` instead the old `defined('TYPO3_MODE') or die();` https://docs.typo3.org/c/typo3/cms-core/main/en-us/Changelog/11.0/Deprecation-92947-DeprecateTYPO3_MODEAndTYPO3_REQUESTTYPEConstants.html#typo3-mode-usage-as-global-script-file-security-gate

Reply:
It does not make sense for v12 because this file goes to `vendor` now and it is outside of the public web directory in any proper installation. As for those, where vendor is inside the public dir, calling any of such file directly will either result in error (which should not be displayed to the user in production), or in a blank page. So no harm in removing it completely.

Reply:
I agree, that it is not needed for v12 and as I see, there is still v11 supported. How can you guarantee it there?

I suggest to branch off a branch for v11 from the current state and use the master only for v12. What's your opinion on that?

Reply:
I stopped using it in my extensions long time ago for these reasons:

> calling any of such file directly will either result in error (which should not be displayed to the user in production), or in a blank page. So no harm in removing it completely.

These constants, as docs say, are exactly to prevent error displaying:

> When those files are located within the public document root of an instance and called via HTTP directly, they may error out and render error messages. This can be a security risk. To prevent this, those files MUST have a security gate as first line. 

If we look into your `ext_localconf.php`, it only calls `\TYPO3\CMS\Core\Utility\ExtensionManagementUtility::addTypoScriptSetup()`, which only sets some global variables. Even if the installation is so insecure that shows errors in production, this file will never show errors.

Same for any other callable file. In my opinion this whole idea of using constants is not worth anything.

But if you are unsure, feel free to add them. No objections from my side. :)